collect_app/src/main/java/org/odk/collect/android/projects/AddProjectDialog.kt

```diff
@@ -60,7 +64,7 @@ class AddProjectDialog : MaterialFullScreenDialogFragment() {
         }
 
         binding.addButton.setOnClickListener {
-            projectsRepository.add(getProjectName(), getProjectIcon(), getProjectColor())
+            projectsRepository.add(Project(uuid.generateUUID(), getProjectName(), getProjectIcon(), getProjectColor()))
```

Comment:
Ach sorry I'd assumed we'd keep the UUID generation in the repository (like how IDs are assigned in the Forms or Instance repository). That way it's ensured that any Project saved to the repo has a unique identifier and no outside caller can mess with that. Happy to discuss if that doesn't make sense.

Reply:
Like this https://github.com/getodk/collect/pull/4501/commits/348d7e8f6d75e13f04184ec47fe610867e7cf17a?